### PR TITLE
make: Add task for formatting hcl/nomad files

### PR DESCRIPTION
**Pull request by endocrimes**
This will automatically format hcl samples during dev, similarly to
changelogfmt and vendorfmt.

On my machine this task takes 0.1s of wall time, so shouldn't introduce
issues during development.

The makefile change is the first commit, and the second is the
formatting of the remaining files

### CODE REVIEW

File: GNUmakefile [hunk 1]
```diff
@@ -214,22 +215,30 @@ proto:
...
 changelogfmt:
 	@echo "--> Making [GH-xxxx] references clickable..."
 	@sed -E 's|([^\[])\[GH-([0-9]+)\]|\1[[GH-\2](https://github.com/hashicorp/nomad/issues/\2)]|g' CHANGELOG.md > changelog.tmp && mv changelog.tmp CHANGELOG.md
 
+## We skip the terraform directory as there are templated hcl configurations
+## that do not successfully compile without rendering
+.PHONY: hclfmt
+hclfmt:
+	@echo "--> Formatting HCL"
+	@find . -path ./terraform -prune -o -name '*.hcl' -name '*.nomad' -exec hclfmt -w {} +
```
Comment (cgbaker) [inline GNUmakefile]: i think this needs to be:
```
find . -path ./terraform -prune -o \( -name '*.nomad' -o -name '*.hcl' \) -exec hclfmt -w {} +
```

Comment (endocrimes) [reply GNUmakefile]: oh oops (actually just needs the extra -o, not the parens tho)

Comment (cgbaker) [reply GNUmakefile]: it played with it some. it seemed to before-coffee me that without the parens, the implicit `-a` between `exec` and the second `name` takes precedence over the `-o`

Comment (cgbaker) [reply GNUmakefile]: >        Please note that -a when specified implicitly (for example by two
       tests appearing without an explicit operator between them) or
       explicitly has higher precedence than -o.  This means that find .
       -name afile -o -name bfile -print will never print afile.

Comment (endocrimes) [reply GNUmakefile]: huh til

**Pull request merged**